build-tools/packages/build-tools/src/type-test-generator/typeTestUtils.ts

```diff
@@ -0,0 +1,281 @@
+import { getFullTypeName, getNodeTypeData, TypeData } from "../typeValidator/typeData";
+import { PackageJson } from "../common/npmPackage";
+
+// Information about the previous package from the package.json is not needed,
+// but error if it's missing since it's nice to separate errors for the dep missing here vs not installed.
+// This ensures that a critical dependency (the previous package version) is correctly declared in the project's package.json.
+export function ensureDevDependencyExists(
+	packageObject: PackageJson,
+	dependencyName: string,
+): void {
+	const dependencyVersion = packageObject?.devDependencies?.[dependencyName];
+	if (typeof dependencyVersion !== "string") {
+		throw new Error(`Did not find devDependency ${dependencyName} in package.json`);
+	}
+}
+
+/**
+ * Fetches the path of the previous package.json or throws an error if not found.
+ * @param previousBasePath - A string representing the path to the root of a package
+ * @returns
+ */
+export function getPreviousPackageJsonPath(previousBasePath: string): string {
+	const previousPackageJsonPath = path.join(previousBasePath, "package.json");
+	if (!existsSync(previousPackageJsonPath)) {
+		throw new Error(`${previousPackageJsonPath} not found.`);
+	}
+	return previousPackageJsonPath;
+}
+
+/**
+ * Attempts to retrieve  a specified type of rollup file path for type definitions from the API Extractor configuration.
+ * @param {string} rollupType - The type of rollup file path to retrieve (ex: "alpha", "beta", "public").
+ * @returns {string} The path to the type definitions file for the specified rollupType, or undefined it cannot be found.
+ * @throws {Error} If api-extractor config cannot be loaded.
+ */
+export function getTypeRollupPathFromExtractorConfig(
+	rollupType: "alpha" | "beta" | "public" | "untrimmed",
+	previousBasePath: string,
+): string | undefined {
+	try {
+		//Load the api-extractor
+		const extractorConfigOptions = ExtractorConfig.tryLoadForFolder({
+			startingFolder: previousBasePath,
+		});
+		if (!extractorConfigOptions || !extractorConfigOptions.configObject) {
+			console.warn(
+				"API Extractor configuration not found. Falling back to default behavior.",
+			);
+			return undefined;
+		}
+		const apiExtractorConfig = extractorConfigOptions.configObject;
+		if (apiExtractorConfig.dtsRollup) {
+			let rollupPath: string | undefined;
+			if (rollupType === "untrimmed") {
+				rollupPath = apiExtractorConfig.dtsRollup.untrimmedFilePath;
+			} else {
+				rollupPath = apiExtractorConfig.dtsRollup[`${rollupType}TrimmedFilePath`];
+			}
+			if (!rollupPath) {
+				console.warn(`Rollup path for "${rollupType}" not found.`);
+				return undefined;
+			}
+			return rollupPath;
+		} else {
+			console.warn(`dtsRollup configuration not found in the API Extractor configuration.`);
+			return undefined;
+		}
+	} catch (error) {
+		console.error(`Error loading API Extractor configuration: ${error}`);
```

Comment:
You'll get better log formatting using the following pattern instead of string interpolation:
```suggestion
		console.error("Error loading API Extractor configuration:", error);
```